packages/aggregator/src/analytics/aggregateAnalytics/aggregations/offsetPeriod.js

```diff
@@ -47,11 +47,8 @@ const compensateForPeriodOffset = (date, { periodType, offset }) =>
 
 export const getDateRangeForOffsetPeriod = (dateRange, config) => {
   validateOffsetPeriodConfig(config);
-  const { offset } = config;
-  const { startDate, endDate } = dateRange;
-
   return {
-    startDate: offset > 0 ? compensateForPeriodOffset(startDate, config) : startDate,
-    endDate: offset < 0 ? compensateForPeriodOffset(endDate, config) : endDate,
+    startDate: compensateForPeriodOffset(dateRange.startDate, config),
+    endDate: compensateForPeriodOffset(dateRange.endDate, config),
```

Comment:
I realised we should add offset compensation to both dates: if not, we may get more analytics than we want, which doesn't play nice with "most_recent" logic

Eg, for
```js
{
  startDate: '20201123',
  endDate: '20201130',
  aggregation: { type: 'OFFSET_PERIOD', config: { periodType: 'week', offset: 1 }}
}
```

which is equivalent to "fetch my the results for the previous period", we want to get results in `['20201116', '20201123']`, then add 1 week to their period

Reply:
Ahhh this makes total sense, good pickup 👍